### ISSUE TITLE
Crash while getting meal from claypot

### ISSUE THREAD

**Issue by inframodem** [anegostudios/VintageStory-Issues#504]
I've been getting steady crashes from getting a meal from a cooking pot. It's not replicated from what I can can tell as doing the same task right after the crash causes it to function as normal, but I've crashed at least 2 other time while trying to get meals from clay pots.

The most recent one occurred while getting the last meal from a clay pot containing hefty serving of red meat stew.

Here is the logs of all previous crashes:

> Version: v1.12.6 (Stable)3/11/2020 10:22:33 PM: Critical error occurred
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.GameContent.BlockEntityCookedContainer.ServeInto(IPlayer player, ItemSlot slot)
   at _atfqQJs3jF5TiC7cub21O31NBFr._XyEtY26ZjA9sjaod3lo4YV2C8fU(Block , BlockSelection )
   at _atfqQJs3jF5TiC7cub21O31NBFr._LeHT0V2RzssAYSw4z1SsZVv6qFl(Single )
   at _atfqQJs3jF5TiC7cub21O31NBFr._x6GbO858YVIb2ClONU8Z2InFFudA(Single )
   at _CpYdgbNZR5kyaDkgJT5mZbaEFXr._UsZi5ylBADLy2OH5Lo1MUzOq9qb(EnumRenderStage , Single )
   at _CeG6XgwLRetaWBDy7Vuk520qgAQ._UsZi5ylBADLy2OH5Lo1MUzOq9qb(EnumRenderStage , Single )
   at _CeG6XgwLRetaWBDy7Vuk520qgAQ._ukBPmjLVq1Em3v9POQXsw4ulMUE(Single )
   at _WgVFI8rJI44rVQv521oBIjuWBxl._ukBPmjLVq1Em3v9POQXsw4ulMUE(Single )
   at _3bQQdrrneYMg7sUqAp4FsQ2XEfN._lShc2IGpVH0SOA7ydOiSsErkATH(Single )
   at _3bQQdrrneYMg7sUqAp4FsQ2XEfN._k3aLAIz3swGZO8aqd7pDrKnxALc(Single )
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(Object sender, FrameEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp)
   at OpenTK.GameWindow.DispatchRenderFrame()
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second)
   at _TPKNmFO93xHjVpa9EspHS9LXWzH._TsFumLr7TNG7iLJEc5uVYqFSeNG(_qRgBLoQrmfmbiUcRN39BpLhs3mN , String[] )
   at _QHQMScYkRrkBYdQDlv19fyrujIx._TsFumLr7TNG7iLJEc5uVYqFSeNG(ThreadStart )
-------------------------------
>Version: v1.12.13 (Stable)3/30/2020 2:21:03 AM: Critical error occurred
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.GameContent.BlockEntityCookedContainer.ServeInto(IPlayer player, ItemSlot slot)
   at _nsofTQyytMqga9qGagCueLtcO3F._KRgmh85jhrXLLVE2yEls7J4Pzv(Block , BlockSelection )
   at _nsofTQyytMqga9qGagCueLtcO3F._NNPZ1F4kAEHnB4YaZY0cvGD1GNY(Single )
   at _nsofTQyytMqga9qGagCueLtcO3F._n9OSi6V2svaK7nIjeYHXjHQoWle(Single )
   at _EHCucaYl0qOUG2EgjClBvzSqvKs._9VogGBryDsuDHZbaiUn3eIjRFBy(EnumRenderStage , Single )
   at _GNOiZepshQPekcjnYBEc2hMXvRC._9VogGBryDsuDHZbaiUn3eIjRFBy(EnumRenderStage , Single )
   at _GNOiZepshQPekcjnYBEc2hMXvRC._rn7I2hVi9vfWQFaodOleZjHvNYq(Single )
   at _OH7N6kcpBWQCoLIkXfPafpR4ygB._rn7I2hVi9vfWQFaodOleZjHvNYq(Single )
   at _PSEVxpeeHecgXSrIYcVaBFUM4t6._fhp20KQAUxB20yapUQQfrY70GVp(Single )
   at _PSEVxpeeHecgXSrIYcVaBFUM4t6._cqp1SUrwh9zcvnTI54N6MpXHR9f(Single )
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(Object sender, FrameEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp)
   at OpenTK.GameWindow.DispatchRenderFrame()
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second)
   at _MQWSaTJgjTk2HAVhjrFm8MbVpyAA._WaNBcVcPwlkzsxutiEV3wy0RhTq(_xOj36cB1DHG7fxIjI5q1UWDnvgc , String[] )
   at _StAYwqRRhBZ4VGu96xdJt21StPd._WaNBcVcPwlkzsxutiEV3wy0RhTq(ThreadStart )
-------------------------------
>Version: v1.12.14 (Stable)4/6/2020 10:42:04 PM: Critical error occurred
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.GameContent.BlockEntityCookedContainer.ServeInto(IPlayer player, ItemSlot slot)
   at _AwTvpwXV1bHGBB8plvoGTYT5ihD._wUuq3uOLN8Fi0aGOzCtPGuj6U7B(Block , BlockSelection )
   at _AwTvpwXV1bHGBB8plvoGTYT5ihD._qL8MKkGIwomV7BZiVAXExTc2RxF(Single )
   at _AwTvpwXV1bHGBB8plvoGTYT5ihD._YZl5SNyBXUKjgXTZCHtfITnWm3e(Single )
   at _clfmWA0YSY2as1BzXGL1FefIS0s._a4Hmi0OF3M1DtfqsRXVEQNMveaP(EnumRenderStage , Single )
   at _hLaepLRCOckdCEvZTPp1GxfgrdT._a4Hmi0OF3M1DtfqsRXVEQNMveaP(EnumRenderStage , Single )
   at _hLaepLRCOckdCEvZTPp1GxfgrdT._UfbESUwVALAm24tWKSVekibkmyy(Single )
   at _wbw3WsD1ic8cZIS5Ql2MkIqPWmb._UfbESUwVALAm24tWKSVekibkmyy(Single )
   at _s54YxMQ8l5DBeihbmw81nJkPTSU._RT1lBvzfhVzcDZ6jmYewDTkUdvx(Single )
   at _s54YxMQ8l5DBeihbmw81nJkPTSU._EZIJU8BLizIQdIOGgxu5BrkHzpV(Single )
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(Object sender, FrameEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp)
   at OpenTK.GameWindow.DispatchRenderFrame()
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second)
   at _gilSYm6aSzOct8TKUm64HAM8OHp._tyidesQGCE1oKHv8YEo8HwYCSty(_RxGTCHor7fqeGRNMEH5pZVfBHHK , String[] )
   at _xmrDk9pogtvR6xvbzmBc2dtFsbu._tyidesQGCE1oKHv8YEo8HwYCSty(ThreadStart )
-------------------------------

This is similar to this issue: https://github.com/anegostudios/VintageStory-Issues/issues/304
However, it doesn't seem to be caused by the same issue as that was resolved in May of last year.

**Comment by Bloodsaw** [comment]
Since you seem to be able to reproduce this crash occasionally I would like to ask you to add some information if it happens again.
1. Please upload your log files from the time directly after a crash (they will be overwritten if you start the game again)
2. Was it in SinglePlayer or MultiPlayer (and if MP, LAN or WAN connection?)
3. Please pay attention and report the steps you've done just before the crash.

**Comment by inframodem** [comment]
For the last instance, the crash occurred during Single Player. As for the other stuff, I'll be sure to write it down when it occurs again.

**Comment by inframodem** [comment]
Took a long time for it to happen again but it happened again.
What occurred was that I had recently eaten a meal then I clicked on the clay pot with one meal left with the recently created empty bowl of the meal I had just eaten.
Maybe the bowl I had in my inventory wasn't a real bowl but a "ghost bowl" since I also noticed that sometimes items still persist in my inventory even if they shouldn't or that multiple items are created even though they aren't actually there. Any ways here is the crash log.

> Running on 64 bit Windows with 16 GB RAM 
Version: v1.12.14 (Stable)6/17/2020 4:21:59 PM: Critical error occurred
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.GameContent.BlockEntityCookedContainer.ServeInto(IPlayer player, ItemSlot slot)
   at _AwTvpwXV1bHGBB8plvoGTYT5ihD._wUuq3uOLN8Fi0aGOzCtPGuj6U7B(Block , BlockSelection )
   at _AwTvpwXV1bHGBB8plvoGTYT5ihD._qL8MKkGIwomV7BZiVAXExTc2RxF(Single )
   at _AwTvpwXV1bHGBB8plvoGTYT5ihD._YZl5SNyBXUKjgXTZCHtfITnWm3e(Single )
   at _clfmWA0YSY2as1BzXGL1FefIS0s._a4Hmi0OF3M1DtfqsRXVEQNMveaP(EnumRenderStage , Single )
   at _hLaepLRCOckdCEvZTPp1GxfgrdT._a4Hmi0OF3M1DtfqsRXVEQNMveaP(EnumRenderStage , Single )
   at _hLaepLRCOckdCEvZTPp1GxfgrdT._UfbESUwVALAm24tWKSVekibkmyy(Single )
   at _wbw3WsD1ic8cZIS5Ql2MkIqPWmb._UfbESUwVALAm24tWKSVekibkmyy(Single )
   at _s54YxMQ8l5DBeihbmw81nJkPTSU._RT1lBvzfhVzcDZ6jmYewDTkUdvx(Single )
   at _s54YxMQ8l5DBeihbmw81nJkPTSU._EZIJU8BLizIQdIOGgxu5BrkHzpV(Single )
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(Object sender, FrameEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp)
   at OpenTK.GameWindow.DispatchRenderFrame()
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second)
   at _gilSYm6aSzOct8TKUm64HAM8OHp._tyidesQGCE1oKHv8YEo8HwYCSty(_RxGTCHor7fqeGRNMEH5pZVfBHHK , String[] )
   at _xmrDk9pogtvR6xvbzmBc2dtFsbu._tyidesQGCE1oKHv8YEo8HwYCSty(ThreadStart )
